singer/utils.py

```diff
@@ -134,6 +134,7 @@ def parse_args(required_config_keys):
     -d,--discover   Run in discover mode
     -p,--properties Properties file: DEPRECATED, please use --catalog instead
     --catalog       Catalog file
+    -D, --dev     Runs the tap in dev mode
```

Comment:
I'd suggest that having a shortcode for this isn't useful since it reduces readability at the savings of 3 characters, maybe if it was `--developer-mode` or something, but this feels like it would just lead to folks not knowing what the `-D` means and copy/pasting.

Reply:
On that note, maybe not critical right now, but the docstring above the changed lines mentions the SPEC and BEST_PRACTICES documents in the getting started repo, so it might be good to eventually update those.

Reply:
Removed the short flag `-D`.